src/main.rs

```diff
@@ -315,34 +312,48 @@ fn main() -> Result<i32> {
         ),
     };
     let pretty = args.pretty.unwrap_or_else(|| buffer.guess_pretty());
-    let mut printer = Printer::new(pretty, args.style, args.stream, buffer);
+    let mut printer = Printer::new(print.clone(), pretty, args.style, args.stream, buffer);
 
-    if print.request_headers {
-        printer.print_request_headers(&request, cookie_jar.clone())?;
-    }
-    if print.request_body {
-        printer.print_request_body(&mut request)?;
-    }
+    printer.print_request_headers(&request, cookie_jar.clone())?;
+    printer.print_request_body(&mut request)?;
 
-    let mut exit_code: i32 = 0;
     if !args.offline {
-        let response = client.execute(request)?;
-        let status = response.status();
-        let check_status = args.check_status.unwrap_or(!args.httpie_compat_mode);
-        exit_code = match status.as_u16() {
-            _ if !(check_status) => 0,
-            300..=399 if !args.follow => 3,
-            400..=499 => 4,
-            500..=599 => 5,
-            _ => 0,
+        let response = if args.follow {
+            let mut client =
+                redirect::RedirectFollower::new(&client, args.max_redirects.unwrap_or(10));
+            if let Some(history_print) = args.history_print {
+                printer.print = history_print;
+            }
```

Comment:
Does this mean `--history-print` applies to the final response?

Reply:
No, we are resetting `printer.print` just before printing the final response.

```rust
        printer.print = print;
        printer.print_response_headers(&response)?;
```

Maybe I should move that line to line 341?

Reply:
Ah, I missed that, even though I looked at that spot specifically. I think it's ok as-is.